### ISSUE TITLE
lua cannot find the socket module

### ISSUE THREAD

**Issue by matnar** [delimitrou/DeathStarBench#20]
I'd like like to use this benchmark, but I'm having some problems with the lua socket module. 

I've launched docker-compose and built files in the wrk2 folder. 
Then, I tried to run the experiments, but I received errors that I'm sending in the enclosed log file.

Apparently, when the ./wrk executable runs, it overrides the lua_path, and the scripts cannot correctly find the lua-socket component. However if I directly run "lua -l socket", it correctly loads the 'socket' module. 
How can I prevent this from happening?

Furthermore, when I run ./wrk, I see that all the requests receive a "non-2xx or 3xx" response; I think this is related to the previous lua-socket problem. Of course, when I open the browser, I can correctly reach the localhost:8080, localhost:8081, and localhost:16686 services. What are the REST APIs invoked, so that I can manually try them using postman?

Thanks
Matteo

[log.txt](https://github.com/delimitrou/DeathStarBench/files/3742748/log.txt)

**Comment by gy1005** [comment]
Could you please check where the lua-socket package installed? The luajit bound with wrk2 should look for packages in 
```
./
/usr/local/share/luajit-2.0.3/
/usr/local/share/lua/5.1/
```

**Comment by matnar** [comment]
Thanks for your reply. On my system, lua and luajit are in `/usr/share` instead of `/usr/local/share`. I tried to create softlink for the two folders, however it still does not work. Where can I update such paths in the wrk code? 

Does the benchmark specifically require those lua/luajit versions?

**Comment by gy1005** [comment]
The simplest solution I can think of is reinstall the lua-socket package with root using luarocks. It will install the packages to `/usr/local`. The other solution you might want to try is to add the path where the  lua-socket package was installed to `LUA_PATH`

We didn't try other versions of lua/luajit. They may or may not work.

**Comment by linnanwang** [comment]
@matnar are you using MAC?

**Comment by matnar** [comment]
Thanks for your replies. I solved by reinstalling the lua-related packages. 

On a system using ubuntu, I launched the following commands: 

`sudo apt install lua5.1`
`sudo apt install luajit`
I installed luarocks as from the official documentation: https://github.com/luarocks/luarocks/wiki/Installation-instructions-for-Unix
`sudo luarocks install luasocket`

As far as I understood, latest versions of lua broke compatibilty. So I installed  packages related to lua 5.1.
